src/TransformableImage.js

```diff
@@ -0,0 +1,205 @@
+import React, { PureComponent, PropTypes } from 'react';
+import { View, Text, Image, ViewPropTypes } from 'react-native';
+import ViewTransformer from 'react-native-view-transformer';
+
+export default class TransformableImage extends PureComponent {
+    static propTypes = {
+        source: PropTypes.oneOfType([
+            PropTypes.shape({ uri: PropTypes.string.isRequired }),
+            PropTypes.number
+        ]).isRequired,
+        style: ViewPropTypes ? ViewPropTypes.style : View.propTypes.style,
+        onLoad: PropTypes.func,
+        onLoadStart: PropTypes.func,
+        dimensions: PropTypes.shape({
```

Comment:
Do you prefer have dimensions as a separate prop?
For gallery images it's in source, not better to keep model the same?

Reply:
You're right, there is some cleaning up to do. If you take a look at line 49, the initial state is still relying on `props.source.dimensions`. I'll change that.

Reply:
```javascript
        image: PropTypes.shape({
            source: PropTypes.oneOfType([
                PropTypes.object,
                PropTypes.number
            ]).isRequired,
            dimensions: PropTypes.shape({ width: PropTypes.number, height: PropTypes.number })
        }).isRequired,
```

I'm renaming `source` to `image` and including `dimensions` in it as you suggested. I had to rename it so we're not accessing the actual source by writing `this.props.source.source`.
It's much clearer.

Reply:
Just to be clear, I think that dimensions should stay in source.

Reply:
I see that i did it wrong for Gallery, dimensions shouldn't be in source object, ahh, then TransformableImage should have source and dimensions (old pikels) separately, my bad.